bugfixtemplates/specifically-thrown-exception/CatchSpecificallyThrownException.java

```diff
@@ -0,0 +1,88 @@
+package bugfixtemplates;
+
+class CatchSpecificallyThrownException {
+
+    void positiveWithoutExplicitCatch() {
+        try {
+            foo();
+            throw new TestException();
+        } catch (IndexOutOfBoundsException e) {
+        } catch (Exception e) {
+        }
+    }
+
+    void negativeWithExplicitCatch() {
+        try {
+            foo();
+            throw new TestException();
+        } catch (IndexOutOfBoundsException e) {
+        } catch (TestException e) {
+        } catch (Exception e) {
+        }
+    }
+
+    void negativeWithoutTryCatch() {
+        foo();
+        throw new TestException();
+    }
+
+    void positiveWithoutRethrowInABranch() {
+        try {
+            foo();
+            throw new TestException();
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Exception [TestException] is explicitly thrown on this line but is caught in the outer catch clause with generic exception on line [34].  Consider adding a catch clause of this specific exception in the outer try statement.